mustermann/lib/mustermann/expander.rb

```diff
@@ -194,8 +194,7 @@ def slice(hash, keys)
 
     def append(uri, values)
       return uri unless values and values.any?
-      entries = values.map { |pair| pair.map { |e| @api_expander.escape(e, also_escape: /[\/\?#\&\=%]/) }.join(?=) }
-      "#{ uri }#{ uri[??]??&:?? }#{ entries.join(?&) }"
+      [uri, QueryStringBuilder.new(values).build].join('?')
```

Comment:
I don't think `QueryStringBuilder` is necessary.  Recursive processing looks fine, but `QueryStringBuilder` is really overkill.